code/game/objects/structures/tables_racks.dm

```diff
@@ -335,6 +393,69 @@
 	log_combat(shover, target, "shoved", "onto [src] (table)")
 	return COMSIG_CARBON_SHOVE_HANDLED
 
+/obj/structure/flippedtable
+	name = "flipped table"
+	desc = "A flipped table."
+	icon = 'icons/obj/flipped_tables.dmi'
+	icon_state = "table"
+	anchored = TRUE
+	density = TRUE
+	layer = ABOVE_MOB_LAYER
+	opacity = FALSE
+	///what type of table the flipped table is
+	var/table_type = /obj/structure/table
+
+/obj/structure/flippedtable/Initialize(mapload)
+	. = ..()
+
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_EXIT = PROC_REF(on_exit),
+	)
+
+	AddElement(/datum/element/connect_loc, loc_connections)
+
+/obj/structure/flippedtable/CanAllowThrough(atom/movable/mover, border_dir)
+	. = ..()
+	if(table_type == /obj/structure/table/glass) //Glass table, jolly ranchers pass
+		if(istype(mover) && (mover.pass_flags & PASSGLASS))
+			return TRUE
+	if(istype(mover, /obj/projectile))
+		var/obj/projectile/projectile = mover
+		//Lets through bullets shot from behind the cover of the table
+		if(projectile.trajectory && angle2dir_cardinal(projectile.trajectory.angle) == dir)
+			return TRUE
+		return FALSE
+	if(border_dir == dir)
+		return FALSE
+	return TRUE
+
+/obj/structure/flippedtable/proc/on_exit(datum/source, atom/movable/leaving, direction)
+	SIGNAL_HANDLER
+
+	if(table_type == /obj/structure/table/glass) //Glass table, jolly ranchers pass
+		if(istype(leaving) && (leaving.pass_flags & PASSGLASS))
+			return
+
+	if(istype(leaving, /obj/projectile))
+		return
+
+	if(direction == dir)
+		return COMPONENT_ATOM_BLOCK_EXIT
+
+/obj/structure/flippedtable/CtrlShiftClick(mob/user)
+	. = ..()
+	if(!istype(user) || !user.can_interact_with())
+		return FALSE
+	user.balloon_alert_to_viewers("flipping table upright...")
+	if(do_after(user, max_integrity * 0.25))
+		var/obj/structure/table/new_table = new table_type(loc)
+		new_table.update_integrity(get_integrity())
+		if(custom_materials)
+			new_table.set_custom_materials(custom_materials)
+		user.balloon_alert_to_viewers("table flipped upright")
```

Comment:
don't need balloon alert because the table flipping back over has visual feedback